### PR TITLE
Fixed issue #1007

### PR DESCRIPTION
**Pull request by dashu-baba**
Closed task opens again after few minutes of closing it - intermittently disappears and reappears in Challenge list.

Fixed

### CODE REVIEW

**Review by maxceem — CHANGES_REQUESTED** [review]
Thanks, @dashu-baba current solution has some issues.

1. If we create a task the review has to be set to internal, and reviewer to the user who created the task. But it's not set during task creating it's only shown on the edit page, but actually it's not set in the challenge data on the server, see demo video https://monosnap.com/file/Wqy420RNkFwnQ3cjzfYRUL8YEe2yDg
    - so we have to set review type to the internal and reviwer to the current user during task creation, not during task edition

2. > Hide the start date field and set the startdate = to the current date/time when activating

    At the moment start data is set during opening edit page, so we would update this date many times when we save the challenge. We have to set this date one time during challenge activation when we click `Launch`, `Activate` or `Launch Active` button anywhere:
      - challenge edit page https://monosnap.com/file/EtxZX8NgZvCacjFbAsmbQsgoY6WKO5
      - challenge view page https://monosnap.com/file/QknKZXAHCjIPeUUgSBcy2hIH3Bj5y7
      - challenge list page https://monosnap.com/file/AVIiA3ioOTMmn40yiE4RWwm9BEWqMf

File: src/components/ChallengeEditor/index.js [hunk 1]
```diff
@@ -1100,6 +1103,7 @@ class ChallengeEditor extends Component {
     if (_.isEmpty(challenge)) {
       return <div>Error loading challenge</div>
     }
+    const isTaskType = checkTaskType(challenge.type)
```
Comment (maxceem) [inline src/components/ChallengeEditor/index.js:1106]: @dashu-baba inside this render method we already have variable `isTask` which detect task slightly different, but actually it gives the same result. Let's reuse it inside this method and don't add duplicate `isTaskType` here.

Comment (dashu-baba) [reply src/components/ChallengeEditor/index.js:1106]: @maxceem , Thanks for the update, I will update them accordingly.

Comment (dashu-baba) [reply src/components/ChallengeEditor/index.js:1106]: @maxceem , Could you please share how the request payload should be during creation of a challenge with **reviewerType** & **reviewer**.

Currently in code I can see **reviewerType** is passed  as a prop of **legacy** object. 
I tried to add reviewer to that object. Also, I tried to add reviewType, reviewer in the main object body.
But each time API returning with message "reviewer" is not allowed".

Please provide some support here.

Comment (maxceem) [reply src/components/ChallengeEditor/index.js:1106]: I have replied on the ticket https://github.com/topcoder-platform/work-manager/issues/1007

**Comment by maxceem** [comment]
@dashu-baba I'm checking the PR, meanwhile could you please resolve the merge conflicts with another changes in `cf-jan-2021` branch.

**Comment by dashu-baba** [comment]
Okay.

File: src/components/ChallengeEditor/index.js [hunk 2]
```diff
@@ -355,6 +355,7 @@ class ChallengeEditor extends Component {
           newChallenge[field][index][option.key] = option.name
         }
       }
+      console.log(newChallenge)
```
Comment (maxceem) [inline src/components/ChallengeEditor/index.js:358]: @dashu-baba let's remove all the debugging console logs.

**Review by maxceem — APPROVED** [review]
Thanks, @dashu-baba. The PR works good for me.

I could merge it after resolving merge conflicts and removing console.log.

File: src/components/ChallengeEditor/index.js (hunk 2 above)
Comment (dashu-baba) [reply src/components/ChallengeEditor/index.js:399]: removed

**Comment by dashu-baba** [comment]
Solved the conflict & conflict log issue.

**Comment by maxceem** [comment]
Thanks, @dashu-baba. All good now, merged PR and accepted the fix.

**Pull request merged**